From 71622eca1aa4830f301803ead87dfd49fe28bb1a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 25 Jul 2016 13:05:09 +0200
Subject: [PATCH] workspaces-applet: Really fix panel crash (issue #302)

This reverts commit 9422bb1563ac42759087b64d531898d567748b1e and add the set_size_request line to prevent the crash.
---
 panel/applets/workspaces/WorkspacesApplet.vala | 21 ++-------------------
 1 file changed, 2 insertions(+), 19 deletions(-)

diff --git a/panel/applets/workspaces/WorkspacesApplet.vala b/panel/applets/workspaces/WorkspacesApplet.vala
index 39e023c..7f9fe2e 100644
--- a/panel/applets/workspaces/WorkspacesApplet.vala
+++ b/panel/applets/workspaces/WorkspacesApplet.vala
@@ -21,29 +21,12 @@ public class WorkspacesApplet : Budgie.Applet
 {
 
     Wnck.Pager? widget;
-    Gtk.EventBox? box;
 
     public WorkspacesApplet()
     {
-        box = new Gtk.EventBox();
-        add(box);
-
-        show_all();
-
-        realize.connect_after(()=> {
-            maybe_integrate_pager();
-        });
-    }
-
-    public void maybe_integrate_pager()
-    {
-        if (widget != null) {
-            return;
-        }
-
         widget = new Wnck.Pager();
-
-        box.add(widget);
+        widget.set_size_request(0, 0);
+        add(widget);
         show_all();
     }
 }
